src/components/ScheduleJob/ScheduleJob.js

```diff
@@ -34,111 +28,31 @@ const ScheduleJob = ({ handleRunJob, match, setOpenScheduleJob }) => {
   const startWeek = getWeekStart(decodeLocale(navigator.language))
   const daysOfWeek = getWeekDays(startWeek)
 
-  useEffect(() => {
-    if (
-      (cron[recurringState.scheduleRepeat.activeOption] === '*' ||
-        cron[recurringState.scheduleRepeat.activeOption] === '0') &&
-      recurringState.scheduleRepeat.activeOption !== 'week'
-    ) {
-      setCron(state => {
-        return {
-          ...state,
-          hour:
-            recurringState.scheduleRepeat.activeOption === 'minute' ? '*' : '0',
-          day:
-            recurringState.scheduleRepeat.activeOption === 'month' ? '1' : '*',
-          [recurringState.scheduleRepeat
-            .activeOption]: `*/${recurringState.scheduleRepeat[
-            recurringState.scheduleRepeat.activeOption
-          ].toString()}`
-        }
-      })
-    } else if (
-      recurringState.scheduleRepeat.activeOption === 'week' &&
-      cron.day === '*'
-    ) {
-      setCron(state => ({
-        ...state,
-        day: `*/${recurringState.scheduleRepeat[
-          recurringState.scheduleRepeat.activeOption
-        ].repeat * 7}`
-      }))
-    }
-  }, [cron, recurringState.scheduleRepeat])
-
-  useEffect(() => {
-    if (
-      recurringState.scheduleRepeat.activeOption === 'week' &&
-      cron.week === '*'
-    ) {
-      setCron(state => ({
-        ...state,
-        week: recurringState.scheduleRepeat.week.daysOfTheWeek.reduce(
-          (prev, next, index, arr) => {
-            const indexOfWeekDay = daysOfWeek.find(day => day.id === next).index
-
-            return (prev +=
-              arr.length - 1 === index ? indexOfWeekDay : indexOfWeekDay + ',')
-          },
-          ''
-        )
-      }))
-    }
-  }, [
-    cron.week,
-    daysOfWeek,
-    recurringState.scheduleRepeat.activeOption,
-    recurringState.scheduleRepeat.week.daysOfTheWeek
-  ])
-
-  const generateCronString = cron => {
-    return Object.keys(cron).reduce((prev, next, index, arr) => {
-      return (prev += arr.length - 1 === index ? cron[next] : cron[next] + ' ')
-    }, '')
-  }
-
-  const getRangeInputValue = () => {
-    return recurringState.scheduleRepeat.activeOption === 'week'
-      ? recurringState.scheduleRepeat.week.repeat.toString()
-      : recurringState.scheduleRepeat[
-          recurringState.scheduleRepeat.activeOption
-        ].toString()
-  }
-
   const handleDaysOfWeek = day => {
     const {
       scheduleRepeat: { week }
     } = recurringState
-    let distinctWeek = week.daysOfTheWeek
-    const indexOfWeekDay = daysOfWeek.find(weekDay => weekDay.id === day).index
-
-    if (week.daysOfTheWeek.indexOf(day) === -1) {
-      distinctWeek = week.daysOfTheWeek.concat(day)
-
-      setCron(state => ({
-        ...state,
-        week: state.week
-          .split(',')
-          .concat(indexOfWeekDay)
-          .sort()
-          .join(',')
-      }))
+    let distinctWeek = week
+
+    if (week.indexOf(day) === -1) {
```

Comment:
If `week` and `day` variables` have string values, you could use the more readable `String.prototype.includes`:
```suggestion
    if (!week.includes(day)) {
```